shared/desktop/renderer/index.js

```diff
@@ -168,14 +165,18 @@ function setupHMR(store) {
   }
 
   module.hot &&
-    module.hot.accept(['../../app/main.desktop', '../../app/routes'], () => {
-      const routes = require('../../app/routes')
-      store.dispatch(refreshRouteDef(routes.loginRouteTree, routes.appRouteTree))
-      try {
-        const NewMain = require('../../app/main.desktop').default
-        render(store, NewMain)
-      } catch (_) {}
-    })
+    module.hot.accept(
+      ['../../app/main.desktop', '../../app/routes', '../../app/routes-login', '../../store/configure-store'],
+      () => {
+        const appRouteTree = require('../../app/routes').default
```

Comment:
One thing I was never clear on was whether doing a `require` here for the routes would affect other places that does `import`. For example, let's say I do:

1) Start app with hot-server in a logged out state (so on `loginRouteTree`)
2) Update the app routes, which triggers HMR and `refreshRouteDef` with the new app routes.
3) Finish the login flow, which triggers a switch from `loginRouteTree` to `appRouteTree`.

Does step 3 switch to the old or new `appRouteTree`?

Last time I tried it I didn't get anywhere because it just refreshed the entire app. I wonder if this question is answerable now...

Reply:
Ok, I triggered an HMR changing people page styling while logged out. Logging in showed the people page *without* the changes I made. The route dependency tree is reloaded via these `require`s, but the action trees are unaffected, so they call the switch with the old routes.

Reply:
Thanks! That's what I was afraid of. Seems like too much of an edge case to fix now, but worth making a TODO somewhere.